### ISSUE TITLE
[array-callback-return]: should not throw error for return argument null in filter, some, every

### ISSUE THREAD

**Issue by anikethsaha** [eslint/eslint#13010]
**What rule do you want to change?**

From the [`docs`](https://eslint.org/docs/rules/array-callback-return#top), the example 

```js
var bar = foo.filter(function(x) {
    if (x) {
        return true;
    } else {
        return;
    }
});
```
is considered as error because of no argument to `return`. But for methods like `filter`, `every` , `some` these should not be an error as not argument return means return `null which is considered as `false` in boolean checks`  

![image](https://user-images.githubusercontent.com/26347874/76116356-dc003b00-600f-11ea-9fa5-c280740151a0.png)

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
I think it should be as default

**Please provide some example code that this change will affect:**

```js

// this will be a correct example

var bar = foo.filter(function(x) {
    if (x) {
        return true;
    } else {
        return;
    }
});


```

**What does the rule currently do for this code?**
Shows linting error 

[demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgYXJyYXktY2FsbGJhY2stcmV0dXJuOiBcImVycm9yXCIqL1xuXG52YXIgYmFyID0gZm9vLmZpbHRlcihmdW5jdGlvbih4KSB7XG4gICAgaWYgKHgpIHtcbiAgICAgICAgcmV0dXJuIHRydWU7XG4gICAgfSBlbHNlIHtcbiAgICAgICAgcmV0dXJuO1xuICAgIH1cbn0pOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6NSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

**What will the rule do after it's changed?**

It wont show any error for these kinds of return....which is limited to some particular methods like `filter`, `some` ...

**Are you willing to submit a pull request to implement this change?**

Yes

**Comment by mdjermanovic** [comment]
Hi @anikethsaha, thanks for the issue!

Isn't this already covered by the existing `allowImplicit` option? 

```js
/*eslint array-callback-return: ["error", { "allowImplicit": true }]*/

var bar = foo.filter(function(x) {
    if (x) {
        return true;
    } else {
        return;
    }
});
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgYXJyYXktY2FsbGJhY2stcmV0dXJuOiBbXCJlcnJvclwiLCB7IFwiYWxsb3dJbXBsaWNpdFwiOiB0cnVlIH1dKi9cblxudmFyIGJhciA9IGZvby5maWx0ZXIoZnVuY3Rpb24oeCkge1xuICAgIGlmICh4KSB7XG4gICAgICAgIHJldHVybiB0cnVlO1xuICAgIH0gZWxzZSB7XG4gICAgICAgIHJldHVybjtcbiAgICB9XG59KTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjUsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

**Comment by anikethsaha** [comment]
Yes....sorry My bad...i missed it. Is it possible for making it default for array methods  which expects boolean returns ? like some , filter, every...

**Comment by mdjermanovic** [comment]
> Yes....sorry My bad...i missed it. Is it possible for making it default for array methods which expects boolean returns ? like some , filter, every...

That would be very confusing for users because the same option applies to all methods, so it wouldn't be clear what is option's default value. If it was split into several options, then it would make sense to consider changing the default value for one of them.

**Comment by anikethsaha** [comment]
> That would be very confusing for users because the same option applies to all methods

Agreed there is a point of confusion for this 👍
